Fix some generics warnings in Eclipse

asashour opened the pull request:
Fix some generics warnings in Eclipse.

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/vaadin/framework/8754)
<!-- Reviewable:end -->

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/vaadin/framework?pullRequest=8754) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/vaadin/framework?pullRequest=8754) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/vaadin/framework?pullRequest=8754) it.</sub>

hesara commented:
There are some inconsistencies in that sometimes TestDateField is used with <?, ?> and sometimes with more specific generics. This is not a blocker by itself, but some of the changes introduce unused imports, which would be nice to remove if not made more specific.

---

Reviewed 11 of 14 files at r1.
Review status: 11 of 14 files reviewed at latest revision, 3 unresolved discussions.

---

*[uitest/src/main/java/com/vaadin/tests/components/gridlayout/LayoutAfterHidingError.java, line 19 at r1](https://reviewable.io:443/reviews/vaadin/framework/8754#-KebtDeLI8hU_LUI06NH:-KebtDeLI8hU_LUI06NI:b-a2qe7j) ([raw file](https://github.com/vaadin/framework/blob/32d54ee013b1fcae317cd581740bc9427d2ef87c/uitest/src/main/java/com/vaadin/tests/components/gridlayout/LayoutAfterHidingError.java#L19)):*
> ```Java
> package com.vaadin.tests.components.gridlayout;
> 
> import java.time.LocalDate;
> ```

why are these imports added if using <?, ?> below?

---

*[uitest/src/main/java/com/vaadin/tests/components/table/TableChildMeasurementHint.java, line 3 at r1](https://reviewable.io:443/reviews/vaadin/framework/8754#-KebtKpHVDElytMBiQpo:-KebtKpHVDElytMBiQpp:b-90ad1b) ([raw file](https://github.com/vaadin/framework/blob/32d54ee013b1fcae317cd581740bc9427d2ef87c/uitest/src/main/java/com/vaadin/tests/components/table/TableChildMeasurementHint.java#L3)):*
> ```Java
> package com.vaadin.tests.components.table;
> 
> import java.time.LocalDate;
> ```

see comment for LayoutAfterHidingError

---

*[uitest/src/main/java/com/vaadin/tests/themes/valo/Tables.java, line 18 at r1](https://reviewable.io:443/reviews/vaadin/framework/8754#-KebtZVF0K8L6CXMEWBI:-KebtZVF0K8L6CXMEWBJ:b-90ad1b) ([raw file](https://github.com/vaadin/framework/blob/32d54ee013b1fcae317cd581740bc9427d2ef87c/uitest/src/main/java/com/vaadin/tests/themes/valo/Tables.java#L18)):*
> ```Java
> package com.vaadin.tests.themes.valo;
> 
> import java.time.LocalDate;
> ```

see comment for LayoutAfterHidingError

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/framework/8754#-:-Kebtdvi9wuvPhA42D1h:b-scm55d)*
<!-- Sent from Reviewable.io -->

asashour committed:
Fix some generics warnings in Eclipse

asashour commented:
Thanks for review, you are right, because I added the generic expectations, then replaced with `<?, ?>` since the types are not used.

Now, I updated the PR with removing the unused imports.

Let me know if I should qualify all to `AbstractDateField<LocalDate, DateResolution>`

hesara commented:
Reviewed 5 of 5 files at r2.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/framework/8754)*
<!-- Sent from Reviewable.io -->
